webapp/src/test/java/com/box/l10n/mojito/service/tm/TMTextUnitStatisticServiceTest.java

```diff
@@ -0,0 +1,216 @@
+import org.junit.Rule;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.*;
+
+/**
+ * @author garion
+ */
+public class TMTextUnitStatisticServiceTest extends ServiceTestBase {
+
+    /**
+     * logger
+     */
+    static Logger logger = LoggerFactory.getLogger(TMTextUnitStatisticServiceTest.class);
+
+    @Autowired
+    TMService tmService;
+
+    @Autowired
+    RepositoryService repositoryService;
+
+    @Autowired
+    AssetRepository assetRepository;
+
+    @Autowired
+    AssetService assetService;
+
+    @Autowired
+    TMTextUnitStatisticRepository tmTextUnitStatisticRepository;
+
+    @Autowired
+    TMTextUnitStatisticService tmTextUnitStatisticService;
+
+    @Rule
+    public TestIdWatcher testIdWatcher = new TestIdWatcher();
+
+    Asset asset;
+    Long tmId;
+    Long assetId;
+    Long tmTextUnitId;
+    String tmTextUnitName = "content --- comment";
+    String tmTextContent = "content";
+    String tmTextComment = "comment";
+    Double lastDayEstimatedVolume = 3D;
+    Double lastPeriodEstimatedVolume = 42D;
+    DateTime lastSeenDate = new DateTime(2021, 11, 25, 0, 0);
+
+    private void createTestTextUnitData(Repository repository) {
+        logger.debug("Create data for test");
+
+        asset = assetService.createAssetWithContent(repository.getId(), "test-asset-path.xliff", "test asset content");
+
+        //make sure asset and its relationships are loaded
+        asset = assetRepository.findById(asset.getId()).orElse(null);
+
+        assetId = asset.getId();
+        tmId = repository.getTm().getId();
+
+        TMTextUnit addTMTextUnit = tmService.addTMTextUnit(tmId, assetId, tmTextUnitName, tmTextContent, tmTextComment);
+        tmTextUnitId = addTMTextUnit.getId();
+    }
+
+    private ImportTextUnitStatisticsBody getImportTextUnitStatisticsBody() {
+        ImportTextUnitStatisticsBody statistic = new ImportTextUnitStatisticsBody();
+        statistic.setName(tmTextUnitName);
+        statistic.setContent(tmTextContent);
+        statistic.setComment(tmTextComment);
+        statistic.setLastDayEstimatedVolume(lastDayEstimatedVolume);
+        statistic.setLastPeriodEstimatedVolume(lastPeriodEstimatedVolume);
+        statistic.setLastSeenDate(lastSeenDate);
+        return statistic;
+    }
+
+    @Test
+    public void importStatisticsWorks() throws Exception {
+        Repository repository = repositoryService.createRepository(testIdWatcher.getEntityName("repository"));
+        createTestTextUnitData(repository);
+
+        List<ImportTextUnitStatisticsBody> textUnitStatistics = new ArrayList<>();
+        ImportTextUnitStatisticsBody statistic = getImportTextUnitStatisticsBody();
+        textUnitStatistics.add(statistic);
+
+        // Act
```

Comment:
knit: not sure about the usefulness of "act, assert" comments

Reply:
Removed